src/GraphQL.MicrosoftDI/DynamicScopedSourceStreamResolver.cs

```diff
@@ -20,11 +20,47 @@ public DynamicScopedSourceStreamResolver(ISourceStreamResolver resolver)
     {
         _resolverFunc = async context =>
         {
-            using var scope = context.RequestServicesOrThrow().CreateScope();
-            return await resolver.ResolveAsync(new ScopedResolveFieldContextAdapter<object>(context, scope.ServiceProvider)).ConfigureAwait(false);
+            var scope = context.RequestServicesOrThrow().CreateScope();
+            var scopedContext = new ScopedResolveFieldContextAdapter<object>(context, scope.ServiceProvider);
+            var observable = await resolver.ResolveAsync(scopedContext).ConfigureAwait(false)
+                ?? throw new InvalidOperationException("The source stream resolver returned null.");
+            // keep the service scope alive until the subscription has been disposed
+            return new ObservableMapper(observable, scope);
```

Comment:
The code in this method could be considered either a bug fix or feature, but typically is not necessary when the returned `IObservable` comes from a singleton service and is the same instance returned to other callers.  With `IAsyncEnumerable` / `yield` support, this feature becomes mandatory for `[Scoped]` to function as expected.

Reply:
Firstly, I do not use subscriptions. Secondly, I use GraphType first approach. So... it's rather hard for me to look throught all these changes and track implementation details in mind, but I'll try.

If I understand correctly before these changes scope was ALWAYS destroyed immediately after creation of returned `IObservable` instance. Code of that observable bits **can** use service provider but it most cases this is root service provider so destroying (and even using) scoped provider does not matter. Then you added support for `IAsyncEnumerable` where scoped services **may** be used wider and in that case you lengthen lifetime of scope - the scope now will die immediately after subscription is disposed and not in `async context => {...}` delegate above. Right?

Reply:
That's essentially correct.

Reply:
Are subscriptions guaranteed to be destroyed?

Reply:
Yes.  If not it is a guaranteed memory leak.  Extreme care in the GraphQL.NET Server project was made to ensure proper disposal in any situation.

Reply:
ok